tools/topic_tools/src/throttle.cpp

```diff
@@ -58,6 +62,9 @@ ros::Time g_last_time;
 bool g_use_wallclock;
 bool g_lazy;
 ros::TransportHints g_th;
+ros::WallDuration g_wait_for_subscribers_timeout{ 1.0 };
+ros::WallDuration g_wait_for_subscribers_connect_time{ 1.0 };
```

Comment:
Neither of these two default values seems to replicate the previous behavior. In favor of not changing existing behavior I would argue the defaults should be zero.

Reply:
I believe I only have to set `g_wait_for_subscribers_connect_time` to `0.0`.

The default value for `g_wait_for_subscribers_timeout` won't matter if I default `g_wait_for_subscribers` to empty.